Content.Server/GameObjects/Components/Damage/DamageOnToolInteractComponent.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Content.Server.GameObjects.Components.Interactable;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.Components.Interactable;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization;
+
+namespace Content.Server.GameObjects.Components.Damage
+{
+    [RegisterComponent]
+    class DamageOnToolInteractComponent : Component, IInteractUsing
+    {
+        #pragma warning disable 649
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+        #pragma warning restore 649
+        public override string Name => "DamageOnToolInteract";
+        private ActSystem _actSystem;
+
+        /* Set in YAML */
+        protected int Damage;
+        protected bool UsingWelding;
+        protected bool UsingAnchoring;
+        protected bool UsingCutting;
+        protected bool UsingMultitool;
+        protected bool UsingPrying;
+        protected bool UsingScrewing;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref Damage,            "damage",           0);
+            serializer.DataField(ref UsingWelding,      "usingWelding",      false);
+            serializer.DataField(ref UsingAnchoring,    "usingAnchoring",   false);
+            serializer.DataField(ref UsingCutting,      "usingCutting",     false);
+            serializer.DataField(ref UsingMultitool,    "usingMultitool",   false);
+            serializer.DataField(ref UsingPrying,       "usingPrying",      false);
+            serializer.DataField(ref UsingScrewing,     "usingScrewing",    false);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _actSystem = _entitySystemManager.GetEntitySystem<ActSystem>();
```

Comment:
Typically there's no need to store this in a private field, unless the `EntitySystem` you want is used a lot in your code. It's easier to just use `EntitySystem.Get<ActSystem>()` in place of the private field when you need it.

Reply:
Thanks. I wasn't aware of that function.